release/release_guide.md

```diff
@@ -0,0 +1,478 @@
+
+Export Some Environment variables in the terminal where you are running the release scripts
+
+- export RELEASE_VERSION=<RELEASE_VERSION_TO_BE_PUBLISHED>-incubating
+- export NEXT_VERSION=<NEW_VERSION_IN_MASTER>
+- export RC_NUM=<release_candidate_num_starting_from_1>
+- export RELEASE_BRANCH=release-<RELEASE_VERSION_TO_BE_PUBLISHED>-rc<release_candidate_num_starting_from_1>
+
+Use cut_release_branch.sh to cut a release branch
+
+- Script: [cut_release_branch.sh](https://github.com/apache/incubator-xtable/blob/master/scripts/release/cut_release_branch.sh)
+
+Usage
+
+```shell
+# Cut a release branch
+./release/scripts/cut_release_branch.sh \
+--release=${RELEASE_VERSION} \
+--next_release=${NEXT_VERSION} \
+--rc_num=${RC_NUM}
+# Show help page
+./incubator-xtable/release/scripts/cut_release_branch.sh -h
+```
+
+## Verify that a Release Build Works
+
+Run "mvn -Prelease clean install" to ensure that the build processes are in good shape. // You need to execute this command once you have the release branch in apache/incubator-xtable
+
+```shell
+mvn -Prelease clean install
+```
+
+# Build a release candidate
+
+The core of the release process is the build-vote-fix cycle. Each cycle produces one release candidate. The Release Manager repeats this cycle until the community approves one release candidate, which is then finalized.
+
+Set up a few environment variables to simplify Maven commands that follow. This identifies the release candidate being built. Start with RC_NUM equal to 1 and increment it for each candidate.
+
+1. Generate Source Release: This will create the tarball under incubator-xtable/src_release directory
+    1. git checkout ${RELEASE_BRANCH}
+    2. cd release && ./scripts/create_source_release.sh
+       > If you have multiple gpg keys(gpg --list-keys), then the signing command will take in the first key most likely.
+       > You will release this when it asks for a passphrase in a pop up.
+       > When asked for passphrase, ensure the intended key is the one asked for.
+       >
+       > Command used in script:
+       >
+       > gpg --armor --detach-sig ${RELEASE_DIR}/incubator-xtable-${RELEASE_VERSION}.src.tgz
+       >
+       > To use a specific key: update as follows: // replace FINGERPRINT
+       >
+       > gpg --local-user [FINGERPRINT] --armor --detach-sig ${RELEASE_DIR}/incubator-xtable-${RELEASE_VERSION}.src.tgz
+    3. Verify Source release is signed and buildable
+        1. cd ../src_release
+        2. gpg --verify apache-xtable-${RELEASE_VERSION}.src.tgz.asc apache-xtable-${RELEASE_VERSION}.src.tgz
+        3. git clone git@github.com:apache/incubator-xtable.git incubator-xtable-tmp-clone && cd incubator-xtable-tmp-clone && git checkout ${RELEASE_BRANCH} && cd ..
+        4. tar -zxvf apache-xtable-${RELEASE_VERSION}.src.tgz 
+        5. diff -qr apache-xtable-${RELEASE_VERSION} incubator-xtable-tmp-clone
```

Comment:
@zabetak Added the diff check you had sent in the voting thread as part of the release guide.